### PR TITLE
[stdlib] Add Unicode script data to scalar properties

### PR DESCRIPTION
**Pull request by Azoy**
I made the `script` and `scriptExtensions` return `UInt8` because the `Unicode.Script` enum defined in the regex parser cannot be made SPI in the stdlib due to the fact that we bootstrap the regex parser using the system's SDK (which won't have this SPI symbol). We can teach the parser to store strings instead and define the enum here as SPI for the string processing module to turn it into the enum.

This depends on https://github.com/apple/swift-experimental-string-processing/pull/255

### CODE REVIEW

File: stdlib/public/core/UnicodeSPI.swift [hunk 1]
```diff
@@ -123,3 +125,43 @@ extension Substring {
...
 }
+
+//===----------------------------------------------------------------------===//
+// Unicode.Script
+//===----------------------------------------------------------------------===//
+
+extension Unicode.Scalar.Properties {
+  @_spi(_Unicode)
+  @available(SwiftStdlib 5.7, *)
+  public var script: UInt8 {
```
Comment (lorentey) [inline stdlib/public/core/UnicodeSPI.swift:136]:
```suggestion
  public var _script: UInt8 {
```

File: stdlib/public/core/UnicodeSPI.swift [hunk 2]
```diff
@@ -123,3 +125,43 @@ extension Substring {
...
+    let rawValue = _swift_stdlib_getScript(_scalar.value)
+
+    _internalInvariant(rawValue != .max, "Unknown script rawValue")
+
+    return rawValue
+  }
+
+  @_spi(_Unicode)
+  @available(SwiftStdlib 5.7, *)
+  public var scriptExtensions: [UInt8] {
```
Comment (lorentey) [inline stdlib/public/core/UnicodeSPI.swift:146]:
```suggestion
  public var _scriptExtensions: [UInt8] {
```

**Review by lorentey — CHANGES_REQUESTED** [review]
We must follow the [Leading Underscore Rule](https://github.com/apple/swift/blob/main/docs/StandardLibraryProgrammersManual.md#the-leading-underscore-rule) when naming non-public interfaces, in every stdlib module.

**Comment by Azoy** [comment]
> We must follow the [Leading Underscore Rule](https://github.com/apple/swift/blob/main/docs/StandardLibraryProgrammersManual.md#the-leading-underscore-rule) when naming non-public interfaces, in every stdlib module.

oops forgot, thanks!

**Comment by Azoy** [comment]
@swift-ci please smoke test

**Comment by Azoy** [comment]
@swift-ci please smoke test

File: stdlib/public/stubs/Unicode/UnicodeScalarProps.cpp [hunk 3]
```diff
@@ -427,3 +429,84 @@ __swift_uint8_t _swift_stdlib_getGeneralCategory(__swift_uint32_t scalar) {
...
+  // all in the array. This should never happen because the array represents all
+  // scalars from 0x0 to 0x10FFFF, but if somehow this branch gets reached,
+  // return 255 to indicate a failure.
+  return std::numeric_limits<__swift_uint8_t>::max();
+#endif
+}
+
+SWIFT_RUNTIME_STDLIB_INTERNAL
+const __swift_uint8_t *_swift_stdlib_getScriptExtensions(__swift_uint32_t scalar,
+                                                      __swift_uint8_t *count) {
```
Comment (rxwei) [inline stdlib/public/stubs/Unicode/UnicodeScalarProps.cpp:493]: Seems misindented

**Review by rxwei — APPROVED** [review]

**Comment by Azoy** [comment]
@swift-ci please smoke test

**Review by lorentey — APPROVED** [review]

**Pull request merged**